### ISSUE TITLE
Calling TFSBTools

### ISSUE THREAD

**Issue by akhst7** [Bioconductor/DelayedArray#35]
After updating DelayedArray to the latest(0.66), upon starring R, DelayedArray tried to load TFSBtools which is not in a list of required package as follows;

> Loading required package: TFBSTools
Error: package or namespace load failed for ‘TFBSTools’:
 .onLoad failed in loadNamespace() for 'DelayedArray', details:
  call: initialize(value, ...)

  This message repeats dozen times until it stops.  I'd appreciate if you can suggest a workaround for this.

**Comment by LTLA** [comment]
Loading _DelayedArray_ works fine for me, with no _TFSBtools_ dependency:

```r
library(DelayedArray)
## Loading required package: stats4
## Loading required package: matrixStats
## Loading required package: BiocGenerics
## Loading required package: parallel
## 
## Attaching package: ‘BiocGenerics’
## 
## The following objects are masked from ‘package:parallel’:
## 
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB
## 
## The following objects are masked from ‘package:stats’:
## 
##     IQR, mad, sd, var, xtabs
## 
## The following objects are masked from ‘package:base’:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colMeans,
##     colnames, colSums, dirname, do.call, duplicated, eval, evalq,
##     Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,
##     pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,
##     rowSums, sapply, setdiff, sort, table, tapply, union, unique,
##     unsplit, which, which.max, which.min
## 
## Loading required package: S4Vectors
## 
## Attaching package: ‘S4Vectors’
## 
## The following object is masked from ‘package:base’:
## 
##     expand.grid
## 
## Loading required package: IRanges
## Loading required package: BiocParallel
## 
## Attaching package: ‘DelayedArray’
## 
## The following objects are masked from ‘package:matrixStats’:
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges
## 
## The following objects are masked from ‘package:base’:
## 
##     aperm, apply
## 
## 
sessionInfo()
## R version 3.5.0 Patched (2018-04-30 r74681)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 16.04.5 LTS
## 
## Matrix products: default
## BLAS: /home/cri.camres.org/lun01/Software/R/R-3-5-branch-release/lib/libRblas.so
## LAPACK: /home/cri.camres.org/lun01/Software/R/R-3-5-branch-release/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
## [1] DelayedArray_0.6.6  BiocParallel_1.14.2 IRanges_2.14.12    
## [4] S4Vectors_0.18.3    BiocGenerics_0.26.0 matrixStats_0.54.0 
## 
## loaded via a namespace (and not attached):
## [1] compiler_3.5.0
```

I would guess that your real problem is somewhere else.

**Comment by akhst7** [comment]
This is the error message that I get every time I open R;

> Loading required package: TFBSTools
Error: package or namespace load failed for ‘TFBSTools’:
 .onLoad failed in loadNamespace() for 'DelayedArray', details:
  call: initialize(value, ...)
  error: attempt to apply non-function

This message usually repeats 20times before it stops. 
 This is my .Rprofile;

` source("https://bioconductor.org/biocLite.R")
.First <- function() {
library(devtools)
library(data.table)
library(ggplot2)
library(dtupdate)
cat("\nSuccessfully loaded .Rprofile at", date(), "\n") }
.Last <- function() {
    cat("\nGoodbye at ", date(), "\n")
}`

`> biocValid()

* sessionInfo()

R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin18.0.0 (64-bit)
Running under: macOS  10.14

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils    
[7] datasets  methods   base     

other attached packages:
 [1] h2o_3.20.0.8         Biostrings_2.48.0    XVector_0.20.0      
 [4] IRanges_2.14.12      S4Vectors_0.18.3     BiocGenerics_0.26.0 
 [7] dtupdate_1.5         ggplot2_3.0.0        data.table_1.11.6   
[10] devtools_1.13.6      BiocInstaller_1.30.0

loaded via a namespace (and not attached):
 [1] fastmatch_1.1-0             plyr_1.8.4                 
 [3] lazyeval_0.2.1              splines_3.5.1              
 [5] BiocParallel_1.14.2         GenomeInfoDb_1.16.0        
 [7] digest_0.6.17               GOSemSim_2.6.2             
 [9] GO.db_3.6.0                 magrittr_1.5               
[11] memoise_1.1.0               matrixStats_0.54.0         
[13] colorspace_1.3-2            blob_1.1.1                 
[15] ggrepel_0.8.0               dplyr_0.7.6                
[17] crayon_1.3.4                RCurl_1.95-4.11            
[19] jsonlite_1.5                bindr_0.1.1                
[21] TFMPvalue_0.0.8             glue_1.3.0                 
[23] gtable_0.2.0                zlibbioc_1.26.0            
[25] scales_1.0.0                DOSE_3.6.1                 
[27] DBI_1.0.0                   Rcpp_0.12.18               
[29] bit_1.1-14                  httr_1.3.1                 
[31] fgsea_1.6.0                 pkgconfig_2.0.2            
[33] XML_3.98-1.16               tidyselect_0.2.4           
[35] rlang_0.2.2                 reshape2_1.4.3             
[37] AnnotationDbi_1.42.1        munsell_0.5.0              
[39] tools_3.5.1                 DirichletMultinomial_1.22.0
[41] RSQLite_2.1.1               stringr_1.3.1              
[43] yaml_2.2.0                  bit64_0.9-7                
[45] caTools_1.17.1.1            purrr_0.2.5                
[47] bindrcpp_0.2.2              pbapply_1.3-4              
[49] DO.db_2.9                   compiler_3.5.1             
[51] rstudioapi_0.7              tibble_1.4.2               
[53] stringi_1.2.4               lattice_0.20-35            
[55] Matrix_1.2-14               pillar_1.3.0               
[57] bitops_1.0-6                GenomicRanges_1.32.7       
[59] qvalue_2.12.0               R6_2.2.2                   
[61] gridExtra_2.3               gtools_3.8.1               
[63] assertthat_0.2.0            seqLogo_1.46.0             
[65] withr_2.1.2                 GenomeInfoDbData_1.1.0     
[67] grid_3.5.1                  Biobase_2.40.0             
[69] lubridate_1.7.4            

Library path directories:
  /Volumes/Tabella/Users/akihoji/R_packages/3.4/library 

* Out-of-date packages
              Package        
data.table    "data.table"   
flowWorkspace "flowWorkspace"
htmlwidgets   "htmlwidgets"  
Rcpp          "Rcpp"         
              LibPath                                                
data.table    "/Volumes/Tabella/Users/akihoji/R_packages/3.4/library"
flowWorkspace "/Volumes/Tabella/Users/akihoji/R_packages/3.4/library"
htmlwidgets   "/Volumes/Tabella/Users/akihoji/R_packages/3.4/library"
Rcpp          "/Volumes/Tabella/Users/akihoji/R_packages/3.4/library"
              Installed Built   ReposVer 
data.table    "1.11.6"  "3.5.1" "1.11.8" 
flowWorkspace "3.28.1"  "3.5.0" "3.28.2" 
htmlwidgets   "1.2"     "3.5.0" "1.3"    
Rcpp          "0.12.18" "3.5.1" "0.12.19"
              Repository                                              
data.table    "https://cran.rstudio.com/src/contrib"                  
flowWorkspace "https://bioconductor.org/packages/3.7/bioc/src/contrib"
htmlwidgets   "https://cran.rstudio.com/src/contrib"                  
Rcpp          "https://cran.rstudio.com/src/contrib"                  

update with biocLite()

* Packages too new for Bioconductor version '3.7'

          Version
superheat "1.0.0"
          LibPath                                                
superheat "/Volumes/Tabella/Users/akihoji/R_packages/3.4/library"

downgrade with biocLite("superheat")`

I'd appreciate any suggestions.

**Comment by mtmorgan** [comment]
It sounds somewhat like this https://github.com/Bioconductor/BiocParallel/issues/81 which I am actively investigating.

**Comment by akhst7** [comment]
I just tried "DemoBiocParallelIssue81" and see what.   I got the exact loading error message;

> Loading required package: DemoBiocParallelIssue81
Loading required package: BiocParallel
Error: package or namespace load failed for ‘DemoBiocParallelIssue81’:
 .onLoad failed in loadNamespace() for 'DemoBiocParallelIssue81', details:
  call: initialize(value, ...)
  error: attempt to apply non-function
Error in .requirePackage(package) : 
  unable to find required package ‘DemoBiocParallelIssue81’

I am investigating as to what causes this in my Mac specifically. I have another Mac with similar R(Rstudio) setup, having the same issue.
